pycheribuild/config/compilation_targets.py

```diff
@@ -199,9 +198,15 @@ def riscv_arch_string(self):
         # Use the insane RISC-V arch string to enable CHERI
         if self.is_baremetal():
             # Baremetal/FreeRTOS only supports softfloat
-            arch_string = "rv64imac"
+            if self.target.cpu_architecture == CPUArchitecture.RISCV32:
+                arch_string = "rv32imac"
```

Comment:
Better to pull out the rv32i/rv64i prefix as a base_arch that then gets imac or imafdc appended.